ietf/review/tests_policies.py

```diff
@@ -724,8 +726,11 @@ def test_default_review_rotation_list_uses_assigned_on_date(self):
                          available_reviewers[2:] + [first_reviewer, second_reviewer])
 
     def test_return_reviewer_to_rotation_top(self):
-        # Should do nothing, this is implicit in this policy, no state change is needed.
-        self.policy.return_reviewer_to_rotation_top(self.append_reviewer())
+        reviewer = self.append_reviewer()
+        self.policy.return_reviewer_to_rotation_top(reviewer, False)
+        self.assertFalse(self.reviewer_settings_for(reviewer).request_assignment_next)
```

Comment:
Between 731 and 732, I think you need to make a new reviewer for the test (by repeating what's at 729.
That should happen both here and for the other policy.

Reply:
Why? The first call to return_reviewer_to_rotation_top with parameter False, does not really do anything, it will not modify the request_assignment_next in any way, so it should still be false after that first call. Then on the second call we call it with wanting to be next set to True, and that will cause request_assignment_next to be set in both policies, and then we check that it is now set.

Reply:
so the unit here is badly named now if all we're doing is setting and testing the "assign me next" bit. If the implementation is really _only_ setting that bit and no other queue manipulation is made, then we should rename this. Otherwise, we'd be missing something moving someone who is already at head of queue (rotation top) to the head of the queue.

Reply:
I'll add an issue to make that change later.

Reply:
I agree that it is badly named. I think this is artifact that there is no clearly defined method of moving someone on the top, other than asking to be next. This function is called in two different situations, once when review request is rejected, and in that case I added the checkbox. It is also called when the document is withdrawn, and in that case this function will be called with wants_to_be_first set to true, as this happens when secretary messed up and assigned something that should not have been assigned, thus it is logical that the reviewer will get new assignment as soon as possible. On the other hand we could add similar checkbox to that location, but as that thing is only used by secretary and very rarely anyways, it might not be needed.